tests/test_dsp.py

```diff
@@ -1,47 +1,117 @@
 import pytest
 import numpy as np
 import numpy.testing as npt
-import matplotlib.pyplot as plt
 from unittest import mock
+import copy
 from pyfar import Signal
 from pyfar import dsp
 
+
 def test_phase_rad(sine_plus_impulse_mock):
     """Test the function returning the phase of a signal in radians."""
     phase = dsp.phase(sine_plus_impulse_mock, deg=False, unwrap=False)
     truth = np.angle(sine_plus_impulse_mock.freq)
     npt.assert_allclose(phase, truth, rtol=1e-7)
 
+
 def test_phase_deg(sine_plus_impulse_mock):
     """Test the function returning the phase of a signal in degrees."""
     phase = dsp.phase(sine_plus_impulse_mock, deg=True, unwrap=False)
     truth = np.degrees(np.angle(sine_plus_impulse_mock.freq))
     npt.assert_allclose(phase, truth, rtol=1e-7)
 
+
 def test_phase_unwrap(sine_plus_impulse_mock):
     """Test the function returning the unwrapped phase of a signal."""
     phase = dsp.phase(sine_plus_impulse_mock, deg=False, unwrap=True)
     truth = np.unwrap(np.angle(sine_plus_impulse_mock.freq))
     npt.assert_allclose(phase, truth, rtol=1e-7)
 
+
 def test_phase_deg_unwrap(sine_plus_impulse_mock):
     """Test the function returning the unwrapped phase of a signal in deg."""
     phase = dsp.phase(sine_plus_impulse_mock, deg=True, unwrap=True)
     truth = np.degrees(np.unwrap(np.angle(sine_plus_impulse_mock.freq)))
     npt.assert_allclose(phase, truth, rtol=1e-7)
 
-def test_group_delay(sine_plus_impulse_mock):
+
+def test_group_delay(impulse_mock):
     """Test the function returning the group delay of a signal."""
-    group_delay = dsp.group_delay(sine_plus_impulse_mock)
-    phase = np.unwrap(np.angle(sine_plus_impulse_mock.freq))
-    bin_dist = (sine_plus_impulse_mock.sampling_rate /
-                    sine_plus_impulse_mock.n_samples)
-    truth = np.squeeze((- np.diff(phase,1,-1, prepend=0) /
-                   (bin_dist * 2 * np.pi)))
-    npt.assert_allclose(group_delay, truth, rtol=1e-7)
+    # test single channel signal
+    signal = impulse_mock[0]
+    grp = dsp.group_delay(signal)
+    assert grp.shape == (signal.n_bins, )
+    npt.assert_allclose(grp, 1000 * np.ones(signal.n_bins))
```

Comment:
For the sake of readability it might be meaningful to define the delays (1000, 750 etc.) only one time either in impulse_mock (and let it return these values) or during its call